CalibFormats/HcalObjects/src/HcalDbService.cc

```diff
@@ -223,10 +224,11 @@ bool HcalDbService::convertPedestalWidths(const HcalGenericDetId& fId,
       pedTrueWidth[i] = x;
       continue;
     }
-    // assume QIE is linear in low range and use x1=0 and x2=1
-    // y = (y2-y1) * (x) [do not add any constant, only scale!]
-    float y2 = coder->charge(*shape, 1, i);
-    float y1 = coder->charge(*shape, 0, i);
+    float y = pedestal->getValues()[i];
+    int x1 = (int)std::floor(y);
```

Comment:
Should these be of type `unsigned` to match [HcalQIECoder.h#L26](https://github.com/cms-sw/cmssw/blob/ecf11e6c9bc34871e16a53e34ed4b8e750e2b3f5/CondFormats/HcalObjects/interface/HcalQIECoder.h#L26) and avoid an irrelevant additional cast? Also, please convert these to C++ casts for conformity with [CMS coding rules](http://cms-sw.github.io/cms_coding_rules.html).

Reply:
Thank you for suggestion/reminder. I've modified the code accordingly.